drivers/rtc/rtc_shell.c

```diff
@@ -146,9 +146,37 @@ static char *strptime(const char *s, const char *format, struct tm *tm_time)
 	}
 }
 
+/* Look up a device by some human-readable string identifier. We
+ * always search among device names. If the feature is available, we
+ * search by node label as well.
+ */
+static const struct device *get_rtc_device(char *id)
+{
+	const struct device *dev;
+
+	dev = device_get_binding(id);
+	if (dev != NULL) {
+		return dev;
+	}
+
+#ifdef CONFIG_DEVICE_DT_METADATA
+	dev = device_get_by_dt_nodelabel(id);
+	if (dev != NULL) {
+		return dev;
+	}
+#endif	/* CONFIG_DEVICE_DT_METADATA */
+
+	return NULL;
+}
```

Comment:
why don't we add a shell-level API to obtain a device pointer using either dev name/nodelabel id?

Reply:
yeah that was my point too

Reply:
sounds good! And then imply CONFIG_DEVICE_DT_METADATA in the shell Kconfig?

Reply:
As it stands, placed under a specific subsystem, it looks a bit off to me. We'd end up with the option auto enabling or not by default depending on some random combination of what subsystem are enabled, and what the corresponding contributors and maintainers decided.

If you really want to make it more exposed to random users, how about changing it to `default y if !SHELL_MINIMAL`? Then it would be coupled with the SHELL subsystem, which makes more sense to me.

Reply:
cc @henrikbrixandersen who IIRC cleaned up a bunch of SHELL options to solve a similar problems (incoherent defaults)

Reply:
> If you really want to make it more exposed to random users, how about changing it to default y if !SHELL_MINIMAL? Then it would be coupled with the SHELL subsystem, which makes more sense to me.

A common API enabled for non-minimal shell configurations would be my preferred solution as well.

Reply:
We may even be able to extend the common API to autocomplete device names/nodelabels for a given device API type.

Reply:
> We may even be able to extend the common API to autocomplete device names/nodelabels for a given device API type.

Yeah seen some first of those patches in the wild already https://github.com/zephyrproject-rtos/zephyr/pull/82249/commits/1c5c28bcaa9d0ce8f3f1a2311958ec3e5a75b335

Reply:
> Wouldn't adding `default y if !SHELL_MINIMAL` to `CONFIG_DEVICE_DT_METADATA` be kind of weird since `CONFIG_DEVICE_DT_METADATA` is under the `kernel` Kconfig? This would be the first mention of `SHELL_MINIMAL` outside of shell related Kconfigs

yup, sounds like we should instead have something like:

```
config SHELL_DEVICE_HELPERS
    bool "Shell device helpers"
    imply DEVICE_DT_METADATA
    default y if !SHELL_MINIMAL
```

Reply:
Ok, I opened a new PR (https://github.com/zephyrproject-rtos/zephyr/pull/82698) to add this.